### ISSUE TITLE
New Feature: shorter commit subject for renames

### ISSUE THREAD

**Issue by aqw** [psyinfra/onyo#708]
When `onyo mv` currently performs a rename, it generates a commit subject like the following:
```
ren [1]: groups/dp/Alex Waite -> groups/dp/Alex Stellfox
```

`tig`'s UI summarizes this as:
```
groups/dp/{Alex Waite => Alex Stellfox}
```

I propose that onyo do something very similar for the commit subject:
```
ren [1]: groups/dp/{Alex Waite -> Alex Stellfox}
```

I prefer this because:

1) it's shorter, and space is at a premium in the subject
2) it highlights what part of the path changed

Implementation:
- if there overlap in the path names, use `{}` to group the differences
  - separate with `->`
  - e.g. `groups/{dp/Alex Waite -> psyinf/Alex Stellfox}/`
- if there's no overlap in the path names, don't use any `{}`
  - e.g. `shelf -> bucket` (i.e. current behavior)

Other commit subject related issue: #475

**Comment by TobiasKadelka** [comment]
This idea makes a lot of sense. I really value our subject lines and they are one of the main things I use to quickly identify what action was when performed, and this idea only makes it easier.

And I had to have a look, it also fits into what we already have for a moving (instead of renaming) `onyo mv`:

`mv [1]: repair/workbench/laptop_apple_mbp2021.serial -> shelf`

Here it is also somewhat shortened, we just state the new directory, not the whole destination path again.

**Comment by bpoldrack** [comment]
> I propose that onyo do something very similar for the commit subject:

I agree. And I would suggest to use the exact same pattern of shortening for `mv`. In some sense it only looks similar now in some cases by accident - it doesn't really work that way yet.

**Comment by aqw** [comment]
The pattern works great for `mv` when moving only one asset. Multiple assets won't really work.
